bin/schema_generator.sh

```diff
@@ -170,8 +173,16 @@ function main() {
     # Copy aliased BQ schemas into place
     alias_schemas $ALIASES_PATH .
 
+    cd ..
+
+    # Copy all BQ schemas from schemas/ to appropriate paths under bq/
+    # TODO: Remove the exemption for xfocsp-error-report/xfocsp-error-report
+    # once ops logic has adapated to the bq/ directory structure.
+    find schemas/ -name '*.bq' | \
+        grep -v xfocsp-error-report/xfocsp-error-report | \
```

Comment:
I think we're going to want to not exclude this. With this exclusion we have 0 bq schemas in the `bq` directory matching `xfocsp-error-report`. It really depends on the order in which we move/deprecate things, but in order for the ops pipeline to start using the `bq/` directory I'd expect the following script to have no output:
```sh
#!/bin/bash
cd schemas/
find -name '*.schema.json' -printf '%P\n' |
while read filename ; do
  namespace=$(echo $filename | cut -d/ -f1)
  if [[ $namespace =~ ^heka|metadata|glean|(pioneer-.*)$ ]] ; then
    continue
  fi
  f=$(cat $filename | jq -r .mozPipelineMetadata.bq_dataset_family)
  t=$(cat $filename | jq -r .mozPipelineMetadata.bq_table)
  if ! [[ -f ../bq/$f/${t}.bq ]]; then
    echo $filename
  fi
done
# output 
# xfocsp-error-report/xfocsp-error-report/xfocsp-error-report.4.schema.json
# telemetry/xfocsp-error-report/xfocsp-error-report.4.schema.json
```

Basically we want a 1:1 mapping of `.schema.json` to `.bq` files, and a 1:1 mapping of `.bq` files to tables. Though, even removing this exclusion, the PR will break the 1:1 mapping as well, as we have two xfocsp tables right now (one empty), with the plan to deprecate one of them.

Assuming the plan is to move `xfocsp-error-report` back into the `telemetry` `bq_dataset_family` once table ACLs are in place, some options here are:

1. Deprecate `xfocsp_error_report` fully before switching the ops logic to use the `bq` dir.
1. As  `bq_dataset_family` isn't used in gcp-ingestion or the ops logic yet, we could modify the telemetry xfocsp-error-report `bq_dataset_family` back to telemetry and do no exclusion here, relying on the rerouting special case in code until we've deprecated `xfocsp_error_report`. This decouples the ops logic adoption of `bq` from xfocsp dataset deprecation but leaves the metadata in a potentially confusing state.
1. Implement further special case logic in the ops pipeline when adopting the `bq` structure
We already have to have a [special case](https://github.com/mozilla-services/cloudops-infra/blob/master/projects/data-shared/tf/modules/bigquery/generate_tfjson.sh#L291) for xfocsp rerouting due to the pipeline family mismatch and getting that into metadata is certainly out of scope for this phase. I think it would be reasonable to have extra special case logic during the transition to using `bq/`, but it would be ideal to avoid it.

I don't have a strong preference for any of the options above, and merging this PR as-is shouldn't break anything. I might have a more developed opinion when I'm actually implementing the ops changes.

Reply:
I like option (2) best for now. I've filed https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/630 to remove the custom `bq_table` from `telemetry/xfocsp-error-report`. Once that is merged, I'll remove this exemption and rerun the logic to verify the output is as we expect.